content/master/concepts/composition-functions.md

```diff
@@ -1,1026 +1,699 @@
 ---
 title: Composition Functions
-state: alpha
+state: beta
 alphaVersion: "1.11"
+betaVersion: "1.14"
 weight: 80
-description: "Composition Functions or XFNs allow for complex Composition patches"
+description: "Composition Functions allow you to template resources using general-purpose programming languages"
 aliases: 
   - /knowledge-base/guides/composition-functions
 ---
 
-<!-- TODO: XFN only use read-only filesystem.
-https://github.com/crossplane/crossplane/issues/4182
--->
+Composition functions (or just functions, for short) are custom programs that
+template Crossplane resources. Crossplane calls composition functions to
+determine what resources it should create when you create a composite resource
+(XR). You can write a function to template resources using a general purpose
+programming language like Go or Python. Using a general purpose programming
+language allows a Function to use more advanced logic to template resources,
+like loops and conditionals.
 
-Composition Functions allow you to supplement or replace your Compositions with
-advanced logic not implementable through available patching strategies. 
+You can build a function using general purpose programming languages such as Go
+or Python. The Crossplane community has also built functions that let you
```

Comment:
nit: felt a bit repetative, since you've already said you can use a general purpose programming language like Go or Python two sentences ago

Reply:
True, but here I'm building on that to make the point that functions aren't only interesting/useful to folks who want to write their own. I think it's worth it.